### PR TITLE
WRQ-19326: Removed the usage of ReactDOM.findDomNode()

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
findDomNode is deprecatd in React 18.3 and will be removed in React 19

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
 agate/ContextualPopupDecorator: changed to get ref based on a newly added dummy sibling 'div' node
 agate/Dropdown/DropdownList:  changed to get ref based on a newly added dummy sibling 'div' node in DropdownListBase
 agate/useScroll/ScrollButton:  Refactored ScrollButtonBase as a functional component in order to use useRef() hook. Added a ref based on a newly added dummy sibling 'div' node, which can be then processed and passed to the Button child.
 agate/useScroll/ScrollButtons:  React.findDomNode could be easily removed as the used ref was already a DOM node

Fixed unit tests
Fixed ui tests

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Fixed JSDoc text for agate/Dropdown

### Links
[//]: # (Related issues, references)
WRQ-19326

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian [daniel.stoian@lgepartner.com](mailto:daniel.stoian@lgepartner.com)

### CODE REVIEW

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Review by adrian-cocoara-lgp — APPROVED** [review]

**Comment by juwonjeong** [comment]
@daniel-stoian-lgp, I was in the process of reviewing this PR. However, after merging this PR (https://github.com/enactjs/enact/pull/3243) we found that it affects sandstone and agate. For example, contextualPopupDecorator has two Sibling Refs that causes the button position to be calculated incorrectly. I will attach the patch of the sandstone please check it. Thank you.
https://github.com/enactjs/sandstone/pull/1623

File: useScroll/ScrollButtons.js [hunk 1]
```diff
@@ -135,7 +134,7 @@ const useScrollButtons = (props) => {
 				if (focusableScrollButtons && !Spotlight.getPointerMode()) {
 					consumeEvent(ev);
 					Spotlight.setPointerMode(false);
-					Spotlight.focus(ReactDOM.findDOMNode(oppositeButton.ref)); // eslint-disable-line react/no-find-dom-node
+					Spotlight.focus(oppositeButton.ref);
```
Comment (juwonjeong) [inline useScroll/ScrollButtons.js:137]: @daniel-stoian-lgp, It seems that ScrollButton ref does not work as we expected. 
In agate sampler, when focusableScrollbar set to true and testing with PageUp/Down button does not work properly. Could youe please check it?

Comment (daniel-stoian-lgp) [reply useScroll/ScrollButtons.js:137]: i updated the logic for ScrollButton.js
PAgeUp and PAgeDown between scrollbuttons seems to work now

**Comment by daniel-stoian-lgp** [comment]
I don't understand why codecov fails
On branch develop, coverage is 84.89% https://app.codecov.io/gh/enactjs/agate/tree/develop
On feature/WRQ-19326, code coverage is higher https://app.codecov.io/gh/enactjs/agate/tree/feature%2FWRQ-19326

File: useScroll/ScrollButtons.js (hunk 1 above)
Comment (juwonjeong) [reply useScroll/ScrollButtons.js:137]: It seems that scrollbuttons work properly. Thank you.

**Comment by juwonjeong** [comment]
> I don't understand why codecov fails On branch develop, coverage is 84.89% https://app.codecov.io/gh/enactjs/agate/tree/develop On feature/WRQ-19326, code coverage is higher https://app.codecov.io/gh/enactjs/agate/tree/feature%2FWRQ-19326

It's hard to understand:( I also checked that dev branch coverage is 84.89%.

**Review by juwonjeong — APPROVED** [review]
LGTM

**Pull request merged**